### DISCUSSION TITLE
Custom metadata reader does not get used

### DISCUSSION THREAD

**Discussion by mlindeman30** [linq2db/linq2db#4491] [Q&A]
I am upgrading after a long time from 3.3.0 to 5.4.1 and I have a case where i need to add a custom meta data reader. According to the [documentation ](https://github.com/linq2db/linq2db/wiki/Releases-and-Roadmap#metadata-reader-changes)this is achieved with the following:

`LinqToDB.Mapping.MappingSchema.Default.AddMetadataReader(new LinqToDbProvider(DbContextOptions.IsSite1 ? typeof(ColumnSite1Attribute) : typeof(ColumnSite2Attribute)));`

However, when i try to get data out of my database this reader is is not used. When running it trough while debugging two other schemas were added to the schema list. This results in that my custom schema is last in the line. With the following piece of code in get the getattributes functions it will always get the first item out of the list:

`Schemas[0].MetadataReader?.GetAttributes<T>(type, memberInfo) ?? Array<T>.Empty;`

Did i read the documentation wrong or is there something else i need to do?

**Comment by sdanyliv** [comment]
You can create new mapping schema (once) and register `MetaDatareader`. When creating `DataConnection`, you can add this MappingSchema. Or register via `DataOptions`.

**Reply by mlindeman30** [reply]
This is what i did, that is why i know it is in the schema list. However there are 2 extra schema's in that list and due to the Schema[0] it will only take the first and that is not the custom one. This is the complete code to register the dataconnection options:

`DbContextOptions = new DbContextOptions(connectionString, siteID);
LinqToDB.Mapping.MappingSchema.Default.AddMetadataReader(new LinqToDbProvider(DbContextOptionsBrouwen.IsSite1 ? typeof(ColumnSite1Attribute) : typeof(ColumnSite2Attribute))); `

`public DbContextOptions(string connectionString, string siteID) 
{
    SiteId = siteID;
    IsSite1 = siteID == "6100";
    DbOptions = new DataOptions().UseSqlServer(connectionString, SqlServerVersion.v2012, SqlServerProvider.SystemDataSqlClient);
}`

Is there an example to add it to the DataOptions?

**Comment by mlindeman30** [comment]
The quickest way is to add the default mappingschema to the DataOptions. 

`DbOptions = new DataOptions().UseSqlServer(connectionString, SqlServerVersion.v2012, SqlServerProvider.SystemDataSqlClient)
    .UseMappingSchema(LinqToDB.Mapping.MappingSchema.Default);`

However, you could argue that this is not the cleanest way to go. Imo it is weird that when looking for metadatareaders you always go for Schema[0] which is not necessarily the default schema. Also it is weird that the default schema is not the first in the schema list.
